src/main/java/org/craftercms/studio/impl/v2/upgrade/operations/RenameUpgradeOperation.java

```diff
@@ -24,48 +24,54 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import static org.craftercms.studio.api.v1.util.StudioConfiguration.BLUE_PRINTS_PATH;
 
 /**
  * Implementation of {@link org.craftercms.studio.api.v2.upgrade.UpgradeOperation} that renames the blueprint in the
  * global repository.
  * @author Dejan Brkic
  */
-public class BlueprintRenameFolderUpgradeOperation extends AbstractUpgradeOperation {
+public class RenameUpgradeOperation extends AbstractUpgradeOperation {
 
-    public static final String CONFIG_KEY_BLUEPRINT_FOLDER_NAME = "blueprintFolderName";
-    public static final String CONFIG_KEY_NEW_FOLDER_NAME = "newFolderName";
+    public static final String CONFIG_KEY_SITE = "site";
+    public static final String CONFIG_KEY_OLD_PATH = "oldPath";
+    public static final String CONFIG_KEY_NEW_PATH = "newPath";
 
-    protected String blueprintFolderName;
-    protected String newFolderName;
+    protected String siteId;
+    protected String oldPath;
+    protected String newPath;
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void doInit(final Configuration config) {
-        blueprintFolderName = config.getString(CONFIG_KEY_BLUEPRINT_FOLDER_NAME);
-        newFolderName = config.getString(CONFIG_KEY_NEW_FOLDER_NAME);
+        siteId = config.getString(CONFIG_KEY_SITE);
+        oldPath = config.getString(CONFIG_KEY_OLD_PATH);
+        newPath = config.getString(CONFIG_KEY_NEW_PATH);
     }
 
     @Override
     public void execute(final String site) throws UpgradeException {
         try {
-            Path globalConfigPath = getRepositoryPath(StringUtils.EMPTY);
-            Path blueprintPath = Paths.get(globalConfigPath.toAbsolutePath().toString(),
-                    studioConfiguration.getProperty(BLUE_PRINTS_PATH), blueprintFolderName);
-            File blueprintFolder = blueprintPath.toFile();
-            if (blueprintFolder.exists()) {
-                Path newBlueprintPath = Paths.get(globalConfigPath.toAbsolutePath().toString(),
-                        studioConfiguration.getProperty(BLUE_PRINTS_PATH), newFolderName);
-                File newBlueprintFolder = newBlueprintPath.toFile();
-                FileUtils.deleteDirectory(newBlueprintFolder);
-                blueprintFolder.renameTo(newBlueprintFolder);
+
+            Path oldP = Paths.get(getRepositoryPath(siteId).getParent().toAbsolutePath().toString(), oldPath);
+            Path newP = Paths.get(getRepositoryPath(siteId).getParent().toAbsolutePath().toString(), newPath);
+
+            File oldF = oldP.toFile();
+            File newF = newP.toFile();
+            if (newF.exists()) {
+                FileUtils.forceDelete(newF);
+            }
+            if (oldF.isDirectory()) {
+                FileUtils.moveDirectory(oldF, newF);
+            } else if (oldF.isFile()) {
+                FileUtils.moveFile(oldF, newF);
             }
 
-            commitAllChanges(StringUtils.EMPTY);
+            commitAllChanges(siteId);
```

Comment:
Please use the `site` parameter from the method, the upgrade manager will send the right value

Reply:
I don't want it to be run for all sites

Reply:
it will not run for all sites, the method parameter indicates the current site being upgraded (or will be empty for the global)

Reply:
A current site being upgraded means on startup will run for all sites

Reply:
the upgrade manager has a different pipeline for the existing sites, this one is in the `system` pipeline so it will only run for the global repo